sycamore/docset.py

```diff
@@ -395,6 +395,24 @@ def regex_replace(self, spec: list[tuple[str, str]], **kwargs) -> "DocSet":
         plan = RegexReplace(self.plan, spec, **kwargs)
         return DocSet(self.context, plan)
 
+    def chain(self, cls: Type[Node], **kwargs) -> "DocSet":
```

Comment:
I like this idea I think. We've been debating how to expose transforms without having to modify The DocSet class every time, and this seems like a reasonable approach, though the whole Node/Transform set of interfaces isn't necessarily super easy to use. 

Minor comments: 
- Should it be `Type[Transform]`? That only excludes scans, I think, which don't make sense here anyway. 
- If it were me I would probably call this something like `apply` or `apply_transform`, but I'm not going to fight too hard about naming.

Reply:
Could we call it `transform`?